plugins/pip/pip.plugin.zsh

```diff
@@ -102,10 +108,25 @@ function pipupall {
   pip list --outdated --format freeze | cut -d= -f1 | ${=xargs} pip install --upgrade
 }
 
-# Uninstalled all installed packages
+# Uninstall all installed packages
 function pipunall {
   # non-GNU xargs does not support nor need `--no-run-if-empty`
   local xargs="xargs --no-run-if-empty"
   xargs --version 2>/dev/null | grep -q GNU || xargs="xargs"
   pip list --format freeze | cut -d= -f1 | ${=xargs} pip uninstall
 }
+
+# Install from GitHub repository
+function pipig {
+  pip install "git+https://github.com/$1/$2.git"
+}
+
+# Install from GitHub branch
+function pipigb {
+  pip install "git+https://github.com/$1/$2.git@$3"
+}
+
+# Install from GitHub pull request
+function pipigp {
+  pip install "git+https://github.com/$1/$2.git@refs/pull/$3/head"
+}
```

Comment:
Could you add here a `compdef _pip <nameofthefunction>` for each one so we can get completion?

Reply:
Done.  Please let me know if I did this right ?

Reply:
Perfect!